AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -208,17 +205,20 @@ - (AWSTask *)saveRecord:(NSData *)data
                                    ];
                     if (!result) {
                         AWSLogError(@"SQLite error. [%@]", db.lastError);
-                        error = db.lastError;
-                        return;
+                        dbError = db.lastError;
+                        return [AWSTask taskWithError:dbError];
```

Comment:
Shouldn't this method always return ```AWSTask```?
I thought this was a miss in the previous code.
Thanks for looking into it.